sdcm/cluster.py

```diff
@@ -1877,10 +1879,67 @@ def run_nodetool(self, sub_cmd, args="", options="", ignore_status=False):
+
         return result
 
+    def check_node_health(self):
+        self.check_nodes_status()
+        self.check_schema_version()
+
+    def check_nodes_status(self):
+        # Validate nodes health
+        node_type = 'target' if self.running_nemesis else 'regular'
+        # TODO: improve this part (using get_nodetool_status function from ScyllaCluster)
+        ClusterHealthValidatorEvent(type='start', name='NodesStatus', status=True, node=self.name)
+        try:
+            self.log.info('Status for %snode %s: %s' % (node_type, self.name, self.run_nodetool('status')))
+
+            ClusterHealthValidatorEvent(type='end', name='NodesStatus', status=True, node=self.name)
+
+        except Exception as ex:
+            ClusterHealthValidatorEvent(type='end', name='NodesStatus', status=False,
+                                        node=self.name,
+                                        error="Unable to get nodetool status from '{node}': {ex}".format(**locals()))
+
+    def check_schema_version(self):
+        # Get schema version
+        ClusterHealthValidatorEvent(type='start', name='NodeSchemaVersion', status=True, node=self.name)
+        errors = []
+        try:
+            result = self.run_nodetool('gossipinfo')
+            schema_version = list(set([line.replace('SCHEMA:', '')
+                                       for line in result.stdout.split() if line.startswith('SCHEMA:')]))[0]
+            self.log.info('Gossipinfo schema_version: %s' % (schema_version))
+
+            # Search for nulls in system.peers
+            peers_details = self.run_cqlsh('select peer, data_center, host_id, rack, release_version, '
+                                           'rpc_address, schema_version, supported_features from system.peers',
+                                           split=True)
+
+            for line in peers_details[3:-2]:
+                line_splitted = line.split('|')
+                current_node_ip = line_splitted[0].strip()
+                nulls = [column for column in line_splitted[1:] if column.strip() == 'null']
+
+                if nulls:
+                    errors.append('Found nulls in system.peers on the node %s: %s' % (current_node_ip, peers_details))
+
+                actual_schema_version = line_splitted[6].strip()
+                if actual_schema_version != schema_version:
+                    errors.append('Expected schema version: %s. Wrong schema version found on the '
+                                  'node %s: %s' % (schema_version, current_node_ip, actual_schema_version))
+        except Exception as e:
+            errors.append('Validate schema version failed. Error: {}'.format(e.message))
+
+        if errors:
+            ClusterHealthValidatorEvent(type='end', name='NodeSchemaVersion', status=False,
+                                        node=self.name,
+                                        error='; '.join(errors))
```

Comment:
can we `'\n'.join(errors)`, since it's would be more readable like that ?
i.e. we are aiming for the ease of usage for ourselves here.